src/hipscat/pixel_math/partition_stats.py

```diff
@@ -136,18 +136,19 @@ def generate_destination_pixel_map(histogram, pixel_map):
         dictionary that maps the integer 3-tuple of a pixel at destination order to the set of
         indexes in histogram for the pixels at the original healpix order
     """
-
     non_none_elements = [i for i in pixel_map if i is not None]
     unique_pixels = np.unique(non_none_elements, axis=0)
 
+    order = hp.npix2order(len(histogram))
+
     result = {}
     for pixel in unique_pixels:
-        source_pixels = []
-        for i, source in enumerate(pixel_map):
-            if not source:
-                continue
-            if source[0] == pixel[0] and source[1] == pixel[1] and histogram[i] > 0:
-                source_pixels.append(i)
-        result[tuple(pixel)] = source_pixels
+        explosion_factor = 4 ** (order - int(pixel[0]))
```

Comment:
This would be worth another comment to clarify what it is doing. Specifically I am having some difficulty following this because I am not sure what the values within the pixel tuple are. What do pixel[0] and pixel[1] represent?

Reply:
Oh! I like that.